src/main/java/br/ufrn/imd/valoris/service/ContaService.java

```diff
@@ -20,4 +23,14 @@ public ContaModel cadastrarConta(ContaDTO contaDTO) {
 
         return contaDao.saveConta(conta);
     }
+
+    public ContaModel debitarConta(String numero, @Valid TransacaoDTO transacaoDTO) {
+        ContaModel conta = findByNumeroIfExists(numero);
+        conta.debitar(transacaoDTO.valor());
+        return contaDao.saveConta(conta);
```

Comment:
Como o DAO retorna uma referência para a conta no `Map`, não é necessário salvar o objeto após modificar seus valores, isso é feito automaticamente.
```suggestion
        return conta;
```